src/azul/plugins/metadata/hca/__init__.py

```diff
@@ -403,3 +410,15 @@ def aggregation_stage(self) -> Type[HCAAggregationStage]:
     @property
     def filter_stage(self) -> Type[HCAFilterStage]:
         return HCAFilterStage
+
+    def validate_version(self, version: str) -> None:
+        # This metadata plugin uses a `dcp2_datetime_format` version with 'Z' as
+        # the time zone indication. e.g. '2001-01-01T01:01:01.000001Z'
+        passed = len(version) == 27 and version.endswith('Z')
```

Comment:
This may be correct, but it's not obviously so. The explanation why the fact that a string that is 27 long and passes parse_dcp2_datetime implies that it must have a six digit, zero-padded microsecond part is a little complicated. Also note that `2001-01-01t1:1:1.1Z` with lower case T passes parse_dcp2_datetime. I'd prefer one of two approaches here: one is format(parse(v)) == v the other is applying the regex from the spec.